driver-core/build.gradle

```diff
@@ -37,8 +37,8 @@ def classifiers = ["linux-x86_64", "linux-aarch_64", "osx-x86_64", "osx-aarch_64
 dependencies {
     api project(path: ':bson', configuration: 'default')
     implementation project(path: ':bson-record-codec', configuration: 'default')
-    implementation project(path: ':bson-kotlin', configuration: 'default')
-    implementation project(path: ':bson-kotlinx', configuration: 'default')
+    implementation project(path: ':bson-kotlin', configuration: 'default'), optional
```

Comment:
If we just removed these what would happen? Would the build fail?

Reply:
The driver-core `KotlinCodecProvider` is included as part of the default registry.

Reply:
Right: that class does canary tests for `KotlinSerializerCodecProvider` and `DataClassCodecProvider`.  So optional seems appropriate.